### ISSUE TITLE
--docekr build fails with podman, even with proper podman-docker binding/aliases

### ISSUE THREAD

**Issue by judovana** [adoptium/temurin-build#3738]
**What are you trying to do?**
Enable  builds with `--docker`, on systems wehre is podman with docker aliases/bindings

**Expected behaviour:**
the build in contianer will pass

**Observed behaviour:**
Currently build fails. I had lost the traces but will elaborate.
It were cryptic messages, suspicions is going to privileged operations. But that is just guessing from air now.

**Any other comments:**
Possible help from @sefroberg needed

**Comment by sxa** [comment]
Yeah this is something that won't have been tested, but would definitely be good to have working. We've made some other changes to allow things to work on podman (Including my recent work on s390x builds) and I suspect it won't be too difficult. Not sure anyone else has tried the `--docker` option for a while!
